pyo3-macros-backend/src/method.rs

```diff
@@ -571,21 +558,66 @@ impl<'a> FnSpec<'a> {
             CallingConvention::Noargs => quote! {
                 _pyo3::impl_::pymethods::PyMethodDef::noargs(
                     #python_name,
-                    _pyo3::impl_::pymethods::PyCFunction(#wrapper),
+                    _pyo3::impl_::pymethods::PyCFunction({
+                        unsafe extern "C" fn trampoline(
```

Comment:
Would be cool not to have to repeat the (almost) identical signature here and in the `trampoline` fn. But the required macro trickery isn't worth it, likely?

Reply:
I agree with this, however I'd like to defer it to a follow-up PR sometime (there's a fair bit of refactoring which could improve the macro code, and I'd like it not to distract readers of this PR from the behavioural change).